MyPerfectOnboardingApplication/Tests/MyPerfectOnboarding.App.Tests/AppTests.cs

```diff
@@ -0,0 +1,305 @@
+        private IUrlLocator _location;
+        private ITimeGenerator _timeGenerator;
+        private IGuidGenerator _guidGenerator;
+        private IListRepository _listRepository;
+
+        [SetUp]
+        public void Init()
+        {
+            _location = Substitute.For<IUrlLocator>();
+            _timeGenerator = Substitute.For<ITimeGenerator>();
+            _guidGenerator = Substitute.For<IGuidGenerator>();
+            _listRepository = Substitute.For<IListRepository>();
+
+            var routeNames = Substitute.For<IControllersRouteNames>();
+            var connectionDetails = Substitute.For<IConnectionDetails>();
+
+            var unityContainer = new UnityContainer();
+            var container = new Container(unityContainer);
+
+            var dependencyContainerConfig = new DependencyContainerConfig(routeNames, connectionDetails);
+            dependencyContainerConfig.RegisterTypes(container);
+
+            container.Register(_location);
+            container.Register(_timeGenerator);
+            container.Register(_guidGenerator);
+            container.Register(_listRepository);
+
+            _listController = container.UnityContainer.Resolve<ListController>();
+            _listController.Request = new HttpRequestMessage();
+            _listController.Configuration = new HttpConfiguration();
+        }
+
+        [Test]
+        public async Task GetAllItemsThreeTimes()
+        {
+            ListItem[] items =
+            {
+                ListItemBuilder.CreateItem("0B9E6EAF-83DC-4A99-9D57-A39FAF258CAC", "aaaaa", "1589-12-03", "1896-04-07"),
+                ListItemBuilder.CreateItem("11AC59B7-9517-4EDD-9DDD-EB418A7C1644", "dfads", "4568-06-23", "8569-08-24")
+            };
+
+            _listRepository.GetAllItemsAsync().Returns(items);
+
+            var message1 = await _listController.ExecuteAction(controller => controller.GetAsync());
+            message1.TryGetContentValue(out IEnumerable<IListItem> resultItems1);
+
+            var message2 = await _listController.ExecuteAction(controller => controller.GetAsync());
+            message2.TryGetContentValue(out IEnumerable<IListItem> resultItems2);
+
+            var message3 = await _listController.ExecuteAction(controller => controller.GetAsync());
+            message3.TryGetContentValue(out IEnumerable<IListItem> resultItems3);
+
+            Assert.Multiple(() => {
+                Assert.That(message1.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+                Assert.That(message2.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+                Assert.That(message3.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+
+                Assert.That(resultItems1, Is.EqualTo(items).Using(ListItemEqualityComparer.Instance));
+                Assert.That(resultItems2, Is.EqualTo(items).Using(ListItemEqualityComparer.Instance));
+                Assert.That(resultItems3, Is.EqualTo(items).Using(ListItemEqualityComparer.Instance));
```

Comment:
It would really be nice to have an extension for the comparer:
```suggestion
                Assert.That(resultItems3, Is.EqualTo(items).UsingListItemComparer());
```